File: maveric-bank-webapp/mavericBank-app/src/app/services/balance.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Balance } from '../modals/balance';
import { environment } from '../../environments/environment.development'; 


@Injectable({
  providedIn: 'root'
})
export class BalanceService {

  constructor(private http: HttpClient) { }

  getBalanceByAccountId(accountId: string): Observable<Balance> {
    return this.http.get<Balance>(`${environment.balanceServiceBaseUrl}/balance/${accountId}`);
  }

  createBalance(balance: Balance): Observable<Balance> {
    return this.http.post<Balance>(`${environment.balanceServiceBaseUrl}/balance`, balance);
  }

  updateBalanceAmount(accountId: string, amountToAdd: number): Observable<Balance> {
    const params = { accountId, amountToAdd: amountToAdd.toString() };
    return this.http.put<Balance>(`${environment.balanceServiceBaseUrl}/balance/update`, null, { params });
  }
}

File: services/balance_Service/src/main/java/com/maverick_shravan/balance_Service/Service/BalanceService.java
package com.maverick_shravan.balance_Service.Service;

import com.maverick_shravan.balance_Service.modal.Balance;

import java.util.Optional;

public interface BalanceService {

    Optional<Balance> getBalanceByAccountId(String accountId);

    Balance addBalance(Balance balance);

    Optional<Balance> updateBalanceAmount(String accountId, double amountToAdd);
}
